src/js/components/sharedComponents/UploadFile.tsx

```diff
@@ -89,10 +96,37 @@ const UploadFile = (): JSX.Element => {
         body: formData
       })
         .then(response => response.json())
-        .catch(e => console.log('error in onDropFile()', e));
+        .catch(e => console.log('fetching error in onDropFile()', e));
+
+      const arcGISResults = geojsonToArcGIS(featureCollection.data.attributes);
+
+      registerGeometry(arcGISResults[0])
+        .then(response => (response.status === 200 ? response.json() : null))
+        .then(results => {
+          if (results) {
+            const oldActiveFeatures = [...activeFeatures];
+            const allGraphics = mapController.generateGraphics(arcGISResults);
+
+            const shapeFileFeatures: LayerFeatureResult = {
+              layerID: 'upload_file_features',
+              layerTitle: 'Upload File Features',
+              sublayerID: null,
+              sublayerTitle: null,
+              features: allGraphics,
+              geoStoreID: results.data.id
+            };
+            oldActiveFeatures.push(shapeFileFeatures);
+            dispatch(setActiveFeatures(oldActiveFeatures));
+          } else {
+            // TODO [ ] - error handling logic
+          }
+        })
+        .catch(e =>
+          console.log('error in registerGeometry() in onDropFile()', e)
+        );
+
+      mapController.processGeojson(arcGISResults);
 
-      const results = geojsonToArcGIS(featureCollection.data.attributes);
-      mapController.processGeojson(results);
       dispatch(toggleTabviewPanel(true));
```

Comment:
I am not sure why we need that action? `selectActiveTab` is already opening the tab?

Reply:
@vaidotasp this accounts for when the user uploads a shapefile/geojson via the pen widget

Reply:
👍